### PR TITLE
remove early stopping/callback for image prediction

### PR DESCRIPTION
**Pull request by annikabrundyn**

### CODE REVIEW

File: src/models/model.py [hunk 1]
```diff
@@ -16,6 +16,32 @@
 
 from pytorch_lightning.metrics.functional import ssim, psnr
 from pytorch_lightning.callbacks.early_stopping import EarlyStopping
+from pytorch_lightning.callbacks import Callback
+
+
+class SavePredImgCallback(Callback):
+    def __init__(self, dl, epoch_logging_freq: int = 50):
+        # save every 50 epochs
```
Comment (annikabrundyn) [inline src/models/model.py:24]: not sure if this is too frequent? basically i want to ability to manually monitor and shut down any jobs but still have the predictions on those same val predictions and if i do every 100 epochs the last prediction might be too far back?

Comment (annikabrundyn) [reply src/models/model.py:24]: thoughts on this?

Comment (jeswan) [reply src/models/model.py:24]: Hmm, might be best to save the best model weights and load the model for generating validation images (I know this is a different paradigm that what we have been using).

File: src/models/model.py [hunk 2]
```diff
@@ -16,6 +16,32 @@
...
+        self.epoch_logging_freq = epoch_logging_freq
+        self.dl = dl
+
+    def on_validation_epoch_end(self, trainer, pl_module):
+        if trainer.current_epoch % self.epoch_logging_freq != 0:
+            return
+
+        batch_idx = 0
+        for img, target, extra in self.dl:
+            img, target, extra = img.cuda(), target.cuda(), extra.cuda()
```
Comment (jeswan) [inline src/models/model.py]:
```suggestion
            img, target, extra = img.to(pl_module.device), target.to(pl_module.device), extra.to(pl_module.device)
```

**Review by jeswan — APPROVED** [review]

**Review by jeswan — CHANGES_REQUESTED** [review]
Let's try this implementation (just one suggestion about `.cuda()`. In the future, we should consider generating the final validation images separately.

File: src/models/model.py (hunk 1 above)
Comment (annikabrundyn) [reply src/models/model.py:24]: i did start with this but its also quite slow/frustrating to do this for 10 models so kind of wanted to generate automatically - I'll think about this again now

**Comment by annikabrundyn** [comment]
yeah I'm happy to commit the script I created to separately load a checkpoint and predict the images, I just wanted to find a way to do it faster than that given how many models we'll be running

**Pull request merged**